assembly/managed/index.ts

```diff
@@ -31,40 +42,90 @@ export class SharedMap<TValue> {
     return message.box!.value;
   }
 
-  set(key: string, value: TValue): void {
+  set(key: string, value: TValue): this {
     let event = new SetSharedMapEvent<TValue>(key, value);
     this.self.send(event);
+    return this;
   }
 
-  delete(key: string): void {
+  delete(key: string): bool {
```

Comment:
Typically speaking, we try to avoid blocking operations, or put them off onto another process. In the case of delete, maybe it makes sense to skip the part where it returns the boolean for `delete()`? That was the reason why I chose that signature in the first place. Everythin else here looks great.